janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -354,7 +359,17 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
                                     fieldDeletions.remove(indexEntry);
                                 }
                             }
-                            deleteIndividualFieldsFromIndex(collectionName, keyIdField, docId, fieldDeletions);
+                            SolrInputDocument doc = new SolrInputDocument();
+                            doc.addField(keyIdField, docId);
+                            for(IndexEntry v: fieldDeletions) {
+                                KeyInformation keyInformation = informations.get(collectionName, v.field);
+                                doc.setField(v.field, keyInformation.getCardinality() == Cardinality.SINGLE?
+                                    fieldDeletes:new HashMap<String, Object>(1) {{ put("remove", convertValue(v.value));}});
```

Comment:
Turns out I can't use ImmutableMap here either because when the solrj client parses the immutablemap, it doesn't like the geopoint.  I've added collections.unmodifiablemap in the appropriate places.
Caused by: org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteSolrException: Error from server at http://127.0.0.1:62442/solr/vertex_shard1_replica1: Unable to parse shape given formats "lat,lon", "x y" or as WKT because java.text.ParseException: Unknown Shape definition [com.google.common.collect.ImmutableMap$Builder:com.google.common.collect.ImmutableMap$Builder@5e7022a2]